keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -101,26 +102,39 @@ class ResourceActuator(
     }
   }
 
+  private suspend fun ResolvableResourceHandler<*, *>.resolve(resource: Resource<out Any>): Pair<Any, Any?> =
+    coroutineScope {
+      val desired = async { desired(resource) }
+      val current = async { current(resource) }
+      desired.await() to current.await()
```

Comment:
What was the `resolve` method on the `ResolvableResourceHandler` interface is now just an extension method here. That way I can always parallelize calls to `desired` and `actual` the same way.